src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -537,7 +553,7 @@ public Database(final StreamInput in) throws IOException {
         public void writeTo(final StreamOutput out) throws IOException {
             out.writeOptionalString(provider);
             out.writeOptionalString(sha256Hash);
-            out.writeOptionalVLong(updatedAt.toEpochMilli());
+            out.writeOptionalVLong(updatedAt == null ? null : updatedAt.toEpochMilli());
```

Comment:
isn't the check for null already encapsulated into method if it does "optional"?

Reply:
Here I call `.toEpochMillis()` method and will will throw NPE if I don't do null check.